src/compositemark/errorbar.ts

```diff
@@ -1,42 +1,395 @@
+import {AggregatedFieldDef, BinTransform, CalculateTransform, TimeUnitTransform} from '../transform';
+import {Flag, keys} from '../util';
+import {Encoding, forEach} from './../encoding';
+import {Field, FieldDef, isContinuous, isFieldDef, PositionFieldDef, vgField} from './../fielddef';
+import * as log from './../log';
+import {GenericUnitSpec, NormalizedLayerSpec, NormalizedUnitSpec} from './../spec';
+import {Orient} from './../vega.schema';
+import {GenericCompositeMarkDef, partLayerMixins} from './common';
 
-export const ERRORBAR: 'error-bar' = 'error-bar';
+export const ERRORBAR: 'errorbar' = 'errorbar';
 export type ErrorBar = typeof ERRORBAR;
 
-export function normalizeErrorBar(spec: GenericUnitSpec<Encoding<Field>, ErrorBar>): NormalizedLayerSpec {
+export type ErrorBarExtent = 'ci' | 'iqr' | 'stderr' | 'stdev';
+export type ErrorBarCenter = 'mean' | 'median';
+
+export type ErrorBarPart = 'bar' | 'line' | 'point' | 'ticks' | 'rule';
+
+const markTypeOfErrorBarPart: { [key: string]: Mark; }= {
+  bar: 'bar',
+  line: 'line',
+  point: 'point',
+  ticks: 'tick',
+  rule: 'rule'
+};
+
+export function getMarkNameOfErrorBarPart(key: ErrorBarPart): Mark {
+  return markTypeOfErrorBarPart[key];
+}
+
+const ERRORBAR_PART_INDEX: Flag<ErrorBarPart> = {
+  bar: 1,
+  line: 1,
+  point: 1,
+  ticks: 1,
+  rule: 1
+};
+
+export const ERRORBAR_PARTS = keys(ERRORBAR_PART_INDEX);
+
+// TODO: Currently can't use `PartsMixins<ErrorBarPart>`
+// as the schema generator will fail
+export type ErrorBarPartsMixins = {
+  [part in ErrorBarPart]?: boolean | MarkConfig
+};
+
+export interface ErrorBarCenterMixins {
+
+  /**
+   * The center of the errorbar. Available options include:
+   * - `"mean": the mean of the data points.
+   * - `"median": the median of the data points.
+   *
+   * __Default value:__ `"mean"`.
+   */
+
+  center?: ErrorBarCenter;
+}
+
+export interface ErrorBarConfig extends ErrorBarPartsMixins, ErrorBarCenterMixins {
+  /**
+   * The options for extent when the center is mean. Available options include:
+   * - `"ci": Extend the rule to the confidence interval of the mean.
+   * - `"stderr": The size of rule are set to the value of standard error, extending from the center.
+   * - `"stdev": The size of rule are set to the value of standard deviation, extending from the center.
+   * - `"iqr": Extend the rule to the q1 and q3.
+   *
+   * __Default value:__ `"stderr"`.
+   */
+  meanExtent?: ErrorBarExtent;
+
+  /**
+   * The options for extent when the center is median. Available options include:
+   * - `"ci": Extend the rule to the confidence interval of the mean.
+   * - `"stderr": The size of rule are set to the value of standard error, extending from the center.
+   * - `"stdev": The size of rule are set to the value of standard deviation, extending from the center.
+   * - `"iqr": Extend the rule to the q1 and q3.
+   *
+   * __Default value:__ `"iqr"`.
+   */
+  medianExtent?: ErrorBarExtent;
```

Comment:
I don't know if we should really add this because it's not useful.

Reply:
I think it might be better to remove these configs and hardcode the default for now.